pareto/utilities/process_data.py

```diff
@@ -0,0 +1,838 @@
+        },
+        requires_at_least_one=[],
+    )
+
+    # Trucking is optional, but if trucking arcs are included, relevant parameters are required
+    trucking_parameters = ["TruckingTime", "TruckingHourlyCost"]
+    trucking_arcs = get_valid_trucking_arc_list()
+
+    if len(set(trucking_arcs) & set(df_parameters)) > 0:
+        # Check that Parameter list contains trucking Parameter tabs
+        missing_trucking_data = set(trucking_parameters) - set(df_parameters)
+        if len(missing_trucking_data) > 0:
+            # Add empty dictionary to df_parameters
+            for s in missing_trucking_data:
+                df_parameters[s] = {}
+            warning_message = (
+                "Trucking arcs are given, but some trucking parameters are missing. The following missing parameters have been set to default values:"
+                + str(missing_trucking_data)
+            )
+            warnings.warn(warning_message)
+
+    # Set Default Data
+    # Iterate through all expected input. For all input left without data, fill with empty dictionary or default data.
+    # raise warning if empty dictionary or default is used.
+    default_used = []
+    # Defaults - Sets
+    all_set_input_tabs = get_valid_input_set_tab_names()
+
+    for set_tab in all_set_input_tabs:
+        if set_tab not in df_sets.keys():
+            df_sets[set_tab] = {}
+            default_used.append(set_tab)
+
+    # Defaults - Parameter
+    # Most default values are defined at the parameter definition. Some input data is not directly
+    # associated with a parameter (e.g. "Economics")
+    default_values = {
+        "Economics": {"discount_rate": 0.08, "CAPEX_lifetime": 20.0},
+    }
+    for input_tab in default_values.keys():
+        if input_tab not in df_parameters.keys():
+            df_parameters[input_tab] = default_values[input_tab]
+            default_used.append(input_tab)
+    # If an empty dictionary is passed to create_model(), the default value for the parameter
+    # is defined at the initialization of the parameter
+    all_parameter_input_tabs = get_valid_input_parameter_tab_names()
+    for param_tab in all_parameter_input_tabs:
+        if param_tab not in df_parameters.keys():
+            df_parameters[param_tab] = {}
+            default_used.append(param_tab)
+    # Raise warning for default values
+    if len(default_used) > 0:
+        warning_message = (
+            f"The following parameters were missing and default values were substituted:"
+            + str(default_used)
+        )
+        warnings.warn(warning_message)
+    return (df_sets, df_parameters)
+
+
+def model_infeasibility_detection(strategic_model):
+    # check_required_data() performs basic feasibility checks:
+    # - at least one source node in the model
+    # - at least on sink node in the model
+    # - each source node has an outgoing arc, each sink node has an incoming arc, and all other nodes have both an incoming and an outgoing arc.
+
```

Comment:
This is a suggestion, if possible plot the network graph using networkx. This will give a good visual representation of the model and make it possible for the user to check if there are any missing arcs without going through the excel sheet. I can help with this implementation. The following resource has a [tutorial](https://networkx.org/documentation/stable/tutorial.html) for the same.

Reply:
I like this! I've created a new issue to track this suggestion.

Reply:
Created issue #321 to be covered in a separate PR.